buildspec_integ.yml

```diff
@@ -1,10 +1,15 @@
 version: 0.2
 
 phases:
+  install:
+    runtime-versions:
+      golang: 1.13
+  pre_build:
+    commands:
+      - mkdir -p /go/src/github.com/aws/amazon-ecs-cli
+      - cp -R . /go/src/github.com/aws/amazon-ecs-cli/
+      - cd /go/src/github.com/aws/amazon-ecs-cli/
   build:
     commands:
-      - latestTag=$(git describe --tags `git rev-list --tags --max-count=1`)
-      - echo "checking out $latestTag"
-      - git checkout $latestTag
       - make integ-test
-      - ./bin/local/ecs-cli.test
\ No newline at end of file
+      - ./bin/local/ecs-cli.test
```

Comment:
How does this buildspec know `./bin/local/ecs-cli.test` must exist? Also, how & where is the test binary built in the pipeline?

Reply:
I'm not sure how we know that about the canary, to be honest, but this succeeds in codebuild


```
[Container] 2020/03/13 03:57:20 Running command ./bin/local/ecs-cli.test
853 | NAME:
854 | ecs-cli - Command line interface for Amazon ECS
855 |  
856 | USAGE:
857 | ecs-cli.test [global options] command [command options] [arguments...]
858 |  
859 | VERSION:
860 | 1.18.1 (*UNKNOWN)
861 |  
862 | AUTHOR:
863 | Amazon Web Services
864 |  
865 | COMMANDS:
866 | configure         Stores a single cluster configuration.
867 | up                Creates the ECS cluster (if it does not already exist) and the AWS resources required to set up the cluster.
868 | down              Deletes the CloudFormation stack that was created by ecs-cli up and the associated resources.
869 | scale             Modifies the number of container instances in your cluster. This command changes the desired and maximum instance count in the Auto Scaling group created by the ecs-cli up command. You can use this command to scale up (increase the number of instances) or scale down (decrease the number of instances) your cluster.
870 | ps                Lists all of the running containers in your ECS cluster.
871 | push              Pushes an image to an Amazon ECR repository.
872 | pull              Pulls an image from an Amazon ECR repository.
873 | images            Lists images from an Amazon ECR repository. Lists all images in all repositories by default.
874 | license           Prints the LICENSE files for the ECS CLI and its dependencies.
875 | compose           Executes docker-compose-style commands on an ECS cluster.
876 | check-attributes  Checks if a given list of container instances can run a given task definition by checking their attributes. Outputs attributes that are required by the task definition but not present on the container instances.
877 | logs              Retrieves container logs from CloudWatch logs. Assumes your Task Definition uses the awslogs driver and has a log stream prefix specified.
878 | registry-creds    Facilitates the creation and use of private registry credentials within ECS.
879 | local             Runs your ECS tasks locally.
880 | help, h           Shows a list of commands or help for one command
881 |  
882 | GLOBAL OPTIONS:
883 | --endpoint value  Use a custom endpoint with the ECS CLI
884 | --help, -h        show help
885 | --version, -v     print the version
886 | PASS
887 | coverage: 2.1% of statements in ./ecs-cli/modules/...
888

```

Reply:
Missed the `make integ-test` statement on the previous line. Please ignore